[Jenkins Operator CI] - Test remote - ClusteringTest on OpenShift

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

The following tests are failing constantly when running on OpenShift:

```
2024-11-10 21:55:13  [ERROR] Errors: 
2024-11-10 21:55:13  [ERROR]   ClusteringTest.testKeycloakCacheIsConnected:243 » ConditionTimeout Assertion condition defined as a Lambda expression in org.keycloak.operator.testsuite.integration.ClusteringTest 
2024-11-10 21:55:13  Expecting any elements of:
2024-11-10 21:55:13    [KeycloakRealmImportStatusCondition{type='Done', status=False, message=''},
2024-11-10 21:55:13      KeycloakRealmImportStatusCondition{type='Started', status=False, message=''},
2024-11-10 21:55:13      KeycloakRealmImportStatusCondition{type='HasErrors', status=True, message='Deployment not yet ready, waiting for it to be ready'}]
2024-11-10 21:55:13  to match given predicate but none did within 10 minutes.
2024-11-10 21:55:13  [ERROR]   ClusteringTest.testKeycloakScaleAsExpected:146 » ConditionTimeout Assertion condition defined as a Lambda expression in org.keycloak.operator.testsuite.utils.K8sUtils found status KeycloakStatusCondition{type='Ready', status=False, message='Waiting for more replicas'} and expected status true within 5 minutes.
2024-11-10 21:55:13  [INFO] 
2024-11-10 21:55:13  [ERROR] Tests run: 125, Failures: 0, Errors: 2, Skipped: 5
```
https://keycloak-jenkins/view/OpenShift/job/rhbk-operator-test-pipeline/893/execution/node/69/log/


### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
@@ -86,7 +86,6 @@ import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
-import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 import java.util.stream.Collectors;
@@ -298,37 +297,21 @@ public class BaseOperatorTest implements QuarkusTestAfterEachCallback {
   public void cleanup() {
       Log.info("Deleting Keycloak CR");
 
-      // due to https://github.com/operator-framework/java-operator-sdk/issues/2314 we
-      // try to ensure that the operator has processed the delete event from root objects
       // this can be simplified to just the root deletion after we pick up the fix
       // it can be further simplified after https://github.com/fabric8io/kubernetes-client/issues/5838
       // to just a timed foreground deletion
       var roots = List.of(Keycloak.class, KeycloakRealmImport.class);
-      var dependents = List.of(StatefulSet.class, Secret.class, Service.class, Pod.class, Job.class);
-
-      var rootsDeleted = CompletableFuture.allOf(roots.stream()
-              .map(c -> k8sclient.resources(c).informOnCondition(List::isEmpty)).toArray(CompletableFuture[]::new));
-      roots.forEach(c -> k8sclient.resources(c).withGracePeriod(10).delete());
+      roots.forEach(c -> k8sclient.resources(c).delete());
+      // enforce that at least the statefulset / pods are gone
       try {
-          rootsDeleted.get(1, TimeUnit.MINUTES);
+          k8sclient
+                  .apps()
+                  .statefulSets()
+                  .inNamespace(namespace)
+                  .withLabels(Constants.DEFAULT_LABELS).informOnCondition(List::isEmpty).get(20, TimeUnit.SECONDS);
       } catch (Exception e) {
-          // delete event should have arrived quickly because this is a background delete
-          throw new RuntimeException(e);
+          throw KubernetesClientException.launderThrowable(e);
       }
-      dependents.stream().map(c -> k8sclient.resources(c).withLabels(Constants.DEFAULT_LABELS))
-              .forEach(r -> r.withGracePeriod(10).delete());
-      // enforce that the dependents are gone
-      Awaitility.await().during(5, TimeUnit.SECONDS).until(() -> {
-          if (dependents.stream().anyMatch(
-                  c -> !k8sclient.resources(c).withLabels(Constants.DEFAULT_LABELS).list().getItems().isEmpty())) {
-              // the operator must have recreated because it hasn't gotten the keycloak
-              // deleted event, keep cleaning
-              dependents.stream().map(c -> k8sclient.resources(c).withLabels(Constants.DEFAULT_LABELS))
-                      .forEach(r -> r.withGracePeriod(0).delete());
-              return false;
-          }
-          return true;
-      });
   }
 
   @Override

--- a/operator/src/test/java/org/keycloak/operator/testsuite/utils/K8sUtils.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/utils/K8sUtils.java
@@ -128,7 +128,7 @@ public final class K8sUtils {
 
     public static CurlResult inClusterCurl(KubernetesClient k8sClient, String namespace, Map<String, String> labels, String... args) {
         Log.infof("Starting cURL in namespace '%s' with labels '%s'", namespace, labels);
-        var podName = "curl-pod-" + UUID.randomUUID();
+        var podName = "curl-pod" + (labels.isEmpty()?"":("-" + UUID.randomUUID()));
         try {
             var builder = new PodBuilder();
             builder.withNewMetadata()
@@ -142,20 +142,22 @@ public final class K8sUtils {
             try {
                 k8sClient.resource(curlPod).create();
             } catch (KubernetesClientException e) {
-                if (e.getCode() != HttpURLConnection.HTTP_CONFLICT) {
+                if (!labels.isEmpty() || e.getCode() != HttpURLConnection.HTTP_CONFLICT) {
                     throw e;
                 }
             }
 
             ByteArrayOutputStream output = new ByteArrayOutputStream();
 
-            try (ExecWatch watch = k8sClient.pods().resource(curlPod).withReadyWaitTimeout(60000)
+            try (ExecWatch watch = k8sClient.pods().resource(curlPod).withReadyWaitTimeout(15000)
                     .writingOutput(output)
                     .exec(Stream.concat(Stream.of("curl"), Stream.of(args)).toArray(String[]::new))) {
-                var exitCode = watch.exitCode().get(15, TimeUnit.SECONDS);
+                var exitCode = watch.exitCode().get(5, TimeUnit.SECONDS);
                 return new CurlResult(exitCode, output.toString(StandardCharsets.UTF_8));
             } finally {
-                k8sClient.resource(curlPod).delete();
+                if (!labels.isEmpty()) {
+                    k8sClient.resource(curlPod).delete();
+                }
             }
         } catch (Exception ex) {
             throw KubernetesClientException.launderThrowable(ex);

